test/functional/plugin/lsp_spec.lua

```diff
@@ -3861,6 +3861,39 @@ describe('LSP', function()
       ]]
       eq(result.method, "initialize")
     end)
+    it('can connect to lsp server via rpc.domain_socket_connect', function()
+      async()
+      exec_lua [[
+        local uv = vim.uv
+        local SOCK = jit.os == "Windows" and "\\\\.\\\\pipe\\pipe.test" or "/tmp/pipe.test"
```

Comment:
Could use something like:

```
local tmpfile_new = helpers.tmpname()
os.remove(tmpfile_new)
```

See https://github.com/neovim/neovim/blob/d92dd2a0c05148154c353f0e7cd2099b2427308a/test/helpers.lua#L371-L396
It handles some more cases

Reply:
Thanks. Do you know if there is something similar for named pipes on Windows? Because they aren't file paths. Or would it be ok to leave the hardcoded pipe name for windows?

Reply:
I'm not aware of anything like that. I guess it's okay to leave it hardcoded.
It's also possible that you'll have to delete the file after calling tmpname to ensure it doesn't exist - to only use it as a means to claim the path in an appropriate location